src/machine/buffer.go

```diff
@@ -4,32 +4,38 @@ import (
 	"runtime/volatile"
 )
 
-const bufferSize = 128
+const defaultRingBufferSize = 128
 
 // RingBuffer is ring buffer implementation inspired by post at
 // https://www.embeddedrelated.com/showthread/comp.arch.embedded/77084-1.php
 type RingBuffer struct {
-	rxbuffer [bufferSize]volatile.Register8
-	head     volatile.Register8
-	tail     volatile.Register8
+	rxbuffer []volatile.Register8
+	head     volatile.Register16
+	tail     volatile.Register16
 }
 
 // NewRingBuffer returns a new ring buffer.
-func NewRingBuffer() *RingBuffer {
-	return &RingBuffer{}
+func NewRingBuffer(size ...int) *RingBuffer {
+	sz := defaultRingBufferSize
+	if len(size) > 0 {
```

Comment:
I believe this `RingBuffer` implementation depends on its size being a power of 2 to support filling _all_ slots in the buffer (although you might not notice an issue until that many slots are filled). 

Might either verify `sz` satisfies that condition, or round up to ensure it does.

Reply:
I agree.
Surely some sort of check is needed.